dirty_cat/super_vectorizer.py

```diff
@@ -130,27 +125,35 @@ class SuperVectorizer(ColumnTransformer):
     columns_: List[Union[str, int]]
         The column names of fitted array.
 
+    types_: Dict[int, type]
+        A learnt mapping of type by column.
+        Key is the column name, value is the final dtype.
+
+    imputed_columns_: List[str]
+        A list of columns for which we imputed the missing values.
+
     """
 
     # Override required parameters
     _required_parameters = []
+    OptionalEstimator = Optional[Union[BaseEstimator, str]]
 
     def __init__(self, *,
                  cardinality_threshold: int = 20,
-                 low_card_str_transformer: Optional[Union[BaseEstimator, str]] = OneHotEncoder(),
-                 high_card_str_transformer: Optional[Union[BaseEstimator, str]] = GapEncoder(),
-                 low_card_cat_transformer: Optional[Union[BaseEstimator, str]] = OneHotEncoder(),
-                 high_card_cat_transformer: Optional[Union[BaseEstimator, str]] = GapEncoder(),
-                 numerical_transformer: Optional[Union[BaseEstimator, str]] = None,
-                 datetime_transformer: Optional[Union[BaseEstimator, str]] = None,
+                 low_card_str_transformer: OptionalEstimator = OneHotEncoder(handle_unknown='ignore'),
+                 high_card_str_transformer: OptionalEstimator = GapEncoder(),
+                 low_card_cat_transformer: OptionalEstimator = OneHotEncoder(handle_unknown='ignore'),
+                 high_card_cat_transformer: OptionalEstimator = GapEncoder(),
+                 numerical_transformer: OptionalEstimator = None,
+                 datetime_transformer: OptionalEstimator = None,
                  auto_cast: bool = True,
+                 impute_missing: str = 'auto',
                  # Following parameters are inherited from ColumnTransformer
-                 handle_missing: str = '',
-                 remainder='passthrough',
-                 sparse_threshold=0.3,
-                 n_jobs=None,
+                 remainder: str = 'passthrough',
+                 sparse_threshold: float = 0.3,
+                 n_jobs: int = None,
```

Comment:
AFAIC, coding standards (PEP8) tell us that in keyword arguments, there should be no space before and after the equal sign:
```suggestion
                 n_jobs: int=None,
```
I am making this comment here, but it apply to many other places.

Reply:
For standard keywords, yes, but not for type hinted ones.
See PEP484, section "Support for singleton types in union", second block for an example:
https://www.python.org/dev/peps/pep-0484/#support-for-singleton-types-in-unions